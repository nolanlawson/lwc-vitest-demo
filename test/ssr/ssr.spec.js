import { renderComponent } from '@lwc/engine-server'
import { describe, it, expect } from 'vitest'
import XButton from 'x/button'

describe('basic ssr test suite', () => {
  it('renders a button on the server side', () => {
    const html = renderComponent('x-button', XButton)

    expect(html).toMatchSnapshot()
  })
})
