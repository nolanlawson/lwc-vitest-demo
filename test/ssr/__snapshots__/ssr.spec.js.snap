// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic ssr test suite > renders a button on the server side 1`] = `
<x-button>
  <template shadowrootmode="open">
    <span>
      <code>
        Is this SSR?
      </code>
      : false
    </span>
    <span>
      <code>
        Was I clicked?
      </code>
      : false
    </span>
    <button class="pretty">
      I am a button!
    </button>
  </template>
</x-button>
`;
